Update configurations / dependencies to the latest versions.

mrvicadai opened the pull request:
## Changelog

- Fix `bsconfig.json` warning by specifying `refmt: 3`
- Use `yarn` as default package manager.
- Update devDependencies to latest versions.
- Default to use `Main.re` instead of `Main.ml`.
- Various formatting.

Commit:
style: reformat rollup.config.js

Commit:
chore: use yarn as the default package manager

Commit:
chore: update devDependencies

Commit:
chore: separate production vs development configuration

Commit:
chore: default to use Main.re in src

Commit:
chore: format bsconfig.json

Commit:
chore: specify refmt: 3 in bsconfig.json

tcoopman commented:
I'm fine with updating the config, but I'm not sure about having a default for `reason`. Can you provide both `ml` and `re` files like in https://github.com/OvermindDL1/bucklescript-tea?

Commit:
chore: revert to default Main.ml in src

mrvicadai commented:
@tcoopman thanks for reviewing! Just addressed the comment by reverting to `Main.ml` in 
https://github.com/tcoopman/bucklescript-tea-starter-kit/pull/3/commits/a2ce86513366d132e574bf63479dd09f9b3b0698

tcoopman commented on .gitignore:
```diff
@@ -25,7 +25,6 @@ lib/
 node_modules
 release/main.js
 .merlin
-yarn.lock
```
I find this a hard thing and will always be subject to preference, so I'll propose that we keep `npm` for now.

mrvicadai replied:
Alright, this is addressed in 
https://github.com/tcoopman/bucklescript-tea-starter-kit/pull/3/commits/5122b40bc08a0216d183136f66dbc7a0cf5a52d5

tcoopman reviewed (changes requested):
Let's revert the yarn stuff for now (so also the lock files). I think this is personal preference, so not really worth the change.

Commit:
chore: remove yarn.lock so people can choose either npm or yarn
